### ISSUE TITLE
Regular GCS uploads are failing with "Unable to detect a Project Id in the current environment."

### ISSUE THREAD

**Issue by MikePikanin** [googleapis/nodejs-common#711]
# Basic info
* Node version: 14.16.1
* @google-cloud/storage version: 5.16.1
* @google-cloud/common version: 3.8.1 (as @google-cloud/storage dependency)

# Issue
When using GCS regular uploads with a client_email and private_key it fails with the following error:
```
Unable to detect a Project Id in the current environment.
To learn more about authentication and Google APIs, visit: 
https://cloud.google.com/docs/authentication/getting-started
```
Though there is no need of setting up projectId explicetly - even Storage class set projectIdRequired to false (https://github.com/googleapis/nodejs-storage/blob/c36525402da8e748971473b1cdd2423e8fd953e1/src/storage.ts#L628)

When upload is resumable (with same credentials) projectId is not required, same should apply for regular upload as well.

Code snippet of upload:
```
            let gcs = require('@google-cloud/storage');
            let storage = new gcs.Storage({credentials: {client_email, private_key}});
            src_stream.pipe(
                storage.bucket(bucket).file(filename)
                .createWriteStream({resumable: false, contentType: content_type}),
```

# Proposal
Instead of:
https://github.com/googleapis/nodejs-common/blob/a7e4ee228291e8ec46f4077ee2d73124c106c94c/src/util.ts#L701
skip resolving projectId if projectIdRequired is false:
```
                !config.projectIdRequired || config.projectId && config.projectId !== '{{projectId}}'
```

**Comment by bcoe** [comment]
@ddelgrosso1 hoping you might be able to give some insight on this one, I know we put quite a bit of work into reworking resumable uploads, wondering if we might have fixed this issue in the process.

**Comment by MikePikanin** [comment]
@bcoe I believe that there is now a redundancy since common was "internalized" recently - https://github.com/googleapis/nodejs-storage/commit/21cbce0459f82a4ad7b7741611cc97c9c3d8a6ca, so the issue needs to be fixed in both places. I might need to open an Issue there now as well.

**Comment by danielbankhead** [comment]
The fix is now available in `nodejs-storage@6.2.0`: https://github.com/googleapis/nodejs-storage/pull/1982

**Comment by MikePikanin** [comment]
@danielbankhead Great job, thanks!
